### PR TITLE
add caasp-ingress-nginx-controller-image

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by kkaempf** [comment]
Where do these `nginx-`* packages come from ?

**Comment by dirkmueller** [comment]
they are taken from the 'repositories' element, which is configured here: https://github.com/SUSE/caasp-container-images/pull/105/files#diff-4e958fc72304fc39d8fb196b7de0d99cR31-R33

**Comment by kkaempf** [comment]
This says `<source path="obsrepositories:/"/>` and isn't helpful.

Let me rephrase the question: Have these packages ever be shipped as part of a released product ? (Do we have ECR allowance to ship them ?)

**Comment by dirkmueller** [comment]
@kkaempf CaaSP v3.0 shipped those exact versions. in CaaSP v4.0 and newer, our documentation referred to the CaaSP v3 image, so 'practically' we shipped it there as well, but most likely never added it to the ECR for 4.0. 

It should have been part of ECR for 3.0, but I guess it doesn't hurt to update it.

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    </type>
+    <version>3.1.0</version>
+    <packagemanager>zypper</packagemanager>
+    <rpm-excludedocs>true</rpm-excludedocs>
+  </preferences>
+  <repository>
+    <source path="obsrepositories:/"/>
+  </repository>
+  <packages type="image">
+    <package name="ingress-nginx-controller"/>
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:35]: The package `nginx-ingress-controller` is used in Devel:CaaSP:5 since a package with the same name was shipped with SLE12SP3. Can this image be updated to reflect this ?

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <specification>ingress-nginx controller</specification>
+  </description>
+  <preferences>
+    <type image="docker" derived_from="obsrepositories:/suse/sle15#15.2">
+      <containerconfig name="ingress-nginx-controller" tag="0.15.0" maintainer="SUSE Containers Team &lt;containers@suse.com&gt;">
+        <entrypoint execute="/usr/bin/ingress-nginx-controller"/>
+        <labels>
+          <suse_label_helper:add_prefix prefix="com.suse.caasp.v5">
+            <label name="com.suse.caasp.v5.description" value="ingress-nginx running on an SLES15 SP2 container guest"/>
+            <label name="com.suse.caasp.v5.title" value="ingress-nginx container"/>
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:16]: Replace 'ingress-nginx-controller` with `nginx-ingress-controller`

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- OBS-AddTag: caasp/v5/ingress-nginx-controller:%%PKG_VERSION%%-rev<VERSION> caasp/v5/ingress-nginx-controller:%%PKG_VERSION%%-rev<VERSION>-build<RELEASE> caasp/v5/ingress-nginx-controller:beta -->
+<image xmlns:suse_label_helper="com.suse.label_helper" schemaversion="6.9" name="ingress-nginx-controller">
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:3]: Replace `ingress-nginx-controller` with `nginx-ingress-controller` similar to the name of the package submitted.

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  <description type="system">
+    <author>SUSE Containers Team</author>
+    <contact>containers@suse.com</contact>
+    <specification>ingress-nginx controller</specification>
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:7]: Replace `ingress-nginx-controller` with `nginx-ingress-controller` similar to the name of the package submitted.

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            <label name="org.opencontainers.image.vendor" value="SUSE LLC"/>
+            <label name="org.opencontainers.image.url" value="https://www.suse.com/products/caas-platform/"/>
+            <label name="org.opencontainers.image.version" value="%%PKG_VERSION%%"/>
+            <label name="org.openbuildservice.disturl" value="%DISTURL%"/>
+            <label name="com.suse.reference" value="registry.suse.com/caasp/v5/ingress-nginx:%%PKG_VERSION%%"/>
+          </suse_label_helper:add_prefix>
+        </labels>
+      </containerconfig>
+    </type>
+    <version>3.1.0</version>
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:27]: The `nginx-ingress-controller` package version is picked up from specification through %%PKG_VERSION%%. This version would only reflect the image revision which can start with `1`.

File: caasp-ingress-nginx-controller-image/_service [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<services>
+    <service name="replace_using_package_version" mode="buildtime">
+        <param name="file">caasp-ingress-nginx-controller-image.kiwi</param>
```
Comment (mmnelemane) [inline caasp-ingress-nginx-controller-image/_service:3]: `nitpic`: May be the name 'caasp-nginx-ingress-controller-image` is better. Just to reflect the package used: https://build.suse.de/request/show/222963

Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/_service:3]: the package naming is incorrect imho. there are two nginx based ingress controllers: 

https://git.k8s.io/ingress-nginx/README.md

this is the one we're intending to use. it is called ingress-nginx controller. thats why I called the image that way. 

There is also a 2nd nginx based controller:


https://www.nginx.com/products/nginx/kubernetes-ingress-controller

This one is called 'NGINX Ingress Controller'. Thats not the one we want to use. hence the renaming of the package to reflect its true origin. 

see https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/ for documentation and links to the origins. we would like to use the kubernetes maintained nginx based ingress controller, also known as 'ingress-nginx'.

**Comment by dirkmueller** [comment]
Regarding the comment on 'ECR' - the ECR does not care about package naming, it cares about package *content*. as such renaming to something obfuscated just to match an earlier ECR doesn't make sense. Otherwise we wouldn't have been allowed to rename cri-o to cri-o-1.18 either to not invalidate ECR.

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi (hunk 5 above)
Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi]: changed version to 1

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi (hunk 4 above)
Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:7]: see https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461407031

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi (hunk 3 above)
Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:3]: see https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461407031

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi (hunk 2 above)
Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:16]: see https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461407031

File: caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi (hunk 1 above)
Comment (dirkmueller) [reply caasp-ingress-nginx-controller-image/caasp-ingress-nginx-controller-image.kiwi:35]: see https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461407031

**Comment by jordimassaguerpla** [comment]
This is unresolvable 

https://build.suse.de/package/show/Devel:CaaSP:5:Containers:CR/caasp-ingress-nginx-controller-image

"nothing provides ingress-nginx-controller"

@dirkmueller The package that was reviewed by legal, security and autobuild was named nginx-ingress-controller, and using the same name makes things easier, meaning the legal bot accepted this package, while it was not accepting the package with the name ingress-nginx-controller.

This is why we have the nginx-ingress-controller package:

https://build.suse.de/package/show/Devel:CaaSP:5/nginx-ingress-controller
https://build.suse.de/package/show/SUSE:SLE-15-SP2:Update:Products:CaaSP:5/nginx-ingress-controller

You pushed us on having the aarch64 story ready asap, so this is why a rename is not a good idea. Could you please follow @mmnelemane advice and use the nginx-ingress-controller package name?

Otherwise, if you **really** want to do a rename, we will introduce uncertainty and delays because of having to ask for legal for a revision.

**Comment by dirkmueller** [comment]
@jordimassaguerpla it was discussed 7 times yesterday already, here's the relevant comment: https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461407031

Please read it, and as you see its only a rename, I think we can work with legal on getting the approval.

**Comment by dirkmueller** [comment]
Also please note that there is a pending SR that is gonna fix this problem: https://build.suse.de/request/show/223099 - maybe you could get it processed?

**Comment by jordimassaguerpla** [comment]
> @jordimassaguerpla it was discussed 7 times yesterday already, here's the relevant comment:
> [#105 (comment)](https://github.com/SUSE/caasp-container-images/pull/105#discussion_r461465971)
> 
> Please read it, and as you see its only a rename, I think we can work with legal on getting the approval. Legal does theoretically not care about the package naming other than that the package naming should be the upstream naming to not introduce new legal hazards (trademark conflicts, for example).

If this is what you want... I just hope you do not come back with hidden commitments like this one https://github.com/SUSE/avant-garde/issues/1843#issuecomment-663827731 and pushing me to rush. You won't have the legal by tomorrow...

My priority is having 4.5, not sure how much time I have left to go after legal on this, but I will do my best.

**Comment by jordimassaguerpla** [comment]
submission for legal bot and autobuild revision

https://build.suse.de/request/show/223112
